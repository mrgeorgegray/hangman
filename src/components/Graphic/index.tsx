import React from "react";

interface GraphicProps {
  chancesRemaining: number;
}

const Graphic: React.FC<GraphicProps> = ({ chancesRemaining }) => (
  <div
    style={{
      width: 160,
      height: 160,
      backgroundColor: "#c3c3c3",
      textAlign: "center",
      padding: 20,
      marginBottom: 20,
    }}
  >
    <p>Hangman graphic @todo...</p>
    <p>(chances remaining: {chancesRemaining})</p>
  </div>
);

export default Graphic;
