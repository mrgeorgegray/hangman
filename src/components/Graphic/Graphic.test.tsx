import React from "react";
import { render } from "@testing-library/react";

import Graphic from ".";

describe("<Graphic />", () => {
  it("renders", () => {
    const { container } = render(<Graphic chancesRemaining={2} />);
    expect(container).toMatchSnapshot();
  });
});
