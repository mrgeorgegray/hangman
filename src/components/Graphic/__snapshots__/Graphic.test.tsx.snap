// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Graphic /> renders 1`] = `
<div>
  <div
    style="width: 160px; height: 160px; background-color: rgb(195, 195, 195); text-align: center; padding: 20px; margin-bottom: 20px;"
  >
    <p>
      Hangman graphic @todo...
    </p>
    <p>
      (chances remaining: 
      2
      )
    </p>
  </div>
</div>
`;
